File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> balance = new Stack<>();

        for (char c : s.toCharArray()){
            try {
                if (c == '('){
                    balance.push(')');
                } else if (c == '[') {
                    balance.push(']');
                } else if (c == '{') {
                    balance.push('}');

                } else {
                    char popped = balance.pop();
                    if (popped != c) {
                        return false;
                    }
                }
            } catch (Exception e) {
                return false;
            }
        }
        return balance.size() > 0 ? false : true;
    }
}

File: 0020-valid-parentheses/0020-valid-parentheses.cs
public class Solution {
    public bool IsValid(string s) {
        Stack cs = new Stack();

        foreach (char c in s.ToCharArray()){
            if (c == '{') cs.Push('}');
            else if (c == '(') cs.Push(')');
            else if (c == '[') cs.Push(']');
            else if (cs.Count > 0 && c != (char)cs.Pop()) return false;
        }
        return cs.Count == 0;
    }
}